source/solvers/vof.cc

```diff
@@ -732,7 +791,268 @@ VOF<dim>::solve_linear_system(const bool initial_step,
+{
+  const double sharpening_threshold =
+    this->simulation_parameters.interface_sharpening.sharpening_threshold;
+  const double interface_sharpness =
+    this->simulation_parameters.interface_sharpening.interface_sharpness;
+
+  FEValues<dim> fe_values_phase_fraction(*this->fs_mapping,
+                                         *this->fe,
+                                         *this->cell_quadrature,
+                                         update_values |
+                                           update_quadrature_points |
+                                           update_JxW_values |
+                                           update_gradients);
+
+
+
+  const unsigned int dofs_per_cell = this->fe->dofs_per_cell;
+  const unsigned int n_q_points    = this->cell_quadrature->size();
+  FullMatrix<double> local_matrix_phase_fraction(dofs_per_cell, dofs_per_cell);
+  Vector<double>     local_rhs_phase_fraction(dofs_per_cell);
+  std::vector<types::global_dof_index> local_dof_indices(dofs_per_cell);
+  std::vector<double>                  phi_phase(dofs_per_cell);
+
+  std::vector<double> phase_values(n_q_points);
+
+  system_rhs_phase_fraction    = 0;
+  system_matrix_phase_fraction = 0;
+
+  for (const auto &cell : this->dof_handler.active_cell_iterators())
+    {
+      if (cell->is_locally_owned())
+        {
+          fe_values_phase_fraction.reinit(cell);
+
+          local_matrix_phase_fraction = 0;
+          local_rhs_phase_fraction    = 0;
+
+          fe_values_phase_fraction.get_function_values(solution, phase_values);
+
+          for (unsigned int q = 0; q < n_q_points; ++q)
+            {
+              auto phase_value = phase_values[q];
+
+              for (unsigned int k = 0; k < dofs_per_cell; ++k)
+                {
+                  phi_phase[k] = fe_values_phase_fraction.shape_value(k, q);
+                }
+              for (unsigned int i = 0; i < dofs_per_cell; ++i)
+                {
+                  // Matrix assembly
+                  for (unsigned int j = 0; j < dofs_per_cell; ++j)
+                    {
+                      local_matrix_phase_fraction(i, j) +=
+                        (phi_phase[j] * phi_phase[i]) *
+                        fe_values_phase_fraction.JxW(q);
+                    }
+
+                  // $$ (if 0 <= \phi <= c)  {\Phi = c ^ (1 - \alpha) * (\phi ^
+                  // \alpha)}$$
+                  // $$ (if c <  \phi <= 1)  {\Phi = 1 - (1 - c) ^ (1 - \alpha)
+                  // * (1 - \phi) ^ \alpha}
+                  if (phase_value >= 0.0 && phase_value <= sharpening_threshold)
+                    local_rhs_phase_fraction(i) +=
+                      std::pow(sharpening_threshold,
+                               (1 - interface_sharpness)) *
+                      std::pow(phase_value, interface_sharpness) *
+                      phi_phase[i] * fe_values_phase_fraction.JxW(q);
+                  else if (phase_value > sharpening_threshold &&
```

Comment:
Why is this an else if? Should it not be an else?

Reply:
Corrected.